File: Back/WebShop-Backend/src/main/java/com/example/WebShop/models/requests/UserUpdateRequest.java
package com.example.WebShop.models.requests;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class UserUpdateRequest {
    @NotBlank
    private String firstName;
    @NotBlank
    private String lastName;
    @NotBlank
    private String currentPassword;
    @NotBlank
    private String newPassword;
    @NotBlank
    @Email
    private String email;
    @NotBlank
    private String phoneNumber;
    @NotBlank
    private String city;
    @NotBlank
    private String avatarUri;
}

File: Front/WebShop-Frontend/src/app/models/userUpdateRequest.ts
export class UserUpdateRequest{
    firstName: string;
    lastName: string;
    currentPassword: string;
    newPassword: string;
    email: string;
    phoneNumber: string;
    city: string;
    avatarUri: string;

    constructor(firstName: string, lastName: string, currentPassword: string, newPassword: string, email: string, phoneNumber: string, city: string, avatarUri: string) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.currentPassword = currentPassword;
        this.newPassword = newPassword;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.city = city;
        this.avatarUri = avatarUri;
    }
}